recipes/gobject-introspection/all/conanfile.py

```diff
@@ -1,129 +1,159 @@
-        else:
-            # https://gitlab.gnome.org/GNOME/gobject-introspection/-/issues/414
-            self.build_requires("meson/0.59.3")
-        self.build_requires("pkgconf/1.7.4")
+        self.tool_requires("meson/[>=1.2.3 <2]")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/[>=2.2 <3]")
         if self.settings.os == "Windows":
-            self.build_requires("winflexbison/2.5.24")
+            self.tool_requires("winflexbison/2.5.25")
         else:
-            self.build_requires("flex/2.6.4")
-            self.build_requires("bison/3.7.6")
-
-    def requirements(self):
-        self.requires("glib/2.73.0")
+            self.tool_requires("flex/2.6.4")
+            self.tool_requires("bison/3.8.2")
+        self.tool_requires("glib/<host_version>")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    def _configure_meson(self):
-        meson = Meson(self)
-        defs = dict()
-        defs["build_introspection_data"] = self.options["glib"].shared
-        defs["datadir"] = os.path.join(self.package_folder, "res")
-
-        meson.configure(
-            source_folder=self._source_subfolder,
-            args=["--wrap-mode=nofallback"],
-            build_folder=self._build_subfolder,
-            defs=defs,
-        )
-        return meson
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+        tc = MesonToolchain(self)
+        if cross_building(self):
+            tc.project_options["gi_cross_use_prebuilt_gi"] = "false"
+        tc.project_options["build_introspection_data"] = self.options.build_introspection_data
+        tc.project_options["datadir"] = "res"
+        tc.generate()
+        deps = PkgConfigDeps(self)
+        deps.generate()
+        # INFO: g-ir-scanner fails to find glib-2.0.pc, so we need to set PKG_CONFIG_PATH
+        env = Environment()
+        env.define("PKG_CONFIG_PATH", self.generators_folder)
+        envvars = env.vars(self)
+        envvars.save_script("pkg_config_env")
+
+    def _patch_sources(self):
+        replace_in_file(self, os.path.join(self.source_folder, "meson.build"),
+                        "subdir('tests')",
+                        "#subdir('tests')")
+        # gir/meson.build expects the gio-unix-2.0 includedir to be passed as a build flag.
+        # Patch this for glib from Conan.
+        replace_in_file(self, os.path.join(self.source_folder, "gir", "meson.build"),
+                        "join_paths(giounix_dep.get_variable(pkgconfig: 'includedir'), 'gio-unix-2.0')",
+                        "giounix_dep.get_variable(pkgconfig: 'includedir')")
```

Comment:
```suggestion
                        "giounix_dep.get_variable(pkgconfig: 'includedir')")
        # Look for data files in res/ instead of share/
        replace_in_file(self, os.path.join(self.source_folder, "tools", "g-ir-tool-template.in"),
                        "os.path.join(filedir, '..', 'share')",
                        "os.path.join(filedir, '..', 'res')")
```
https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/1.78.1/tools/g-ir-tool-template.in#L49